### PR TITLE
checkErrorResponse documented in the tutorial

### PR DESCRIPTION
**Pull request by platan**
This is a follow-up to https://github.com/badges/shields/pull/1670

This change contains:
- up-to-date code example from `server.js`
- `checkErrorResponse` function mentioned in tutorial
- static badges as example of errors (https://img.shields.io/docker/automated/,/ubuntu-remix.svg is no longer "invalid")
- up-to-date links to `server.js`

### CODE REVIEW

File: doc/TUTORIAL.md [hunk 1]
```diff
@@ -121,22 +121,21 @@ cache(function(data, match, sendBadge, request) {
...
+      var is_automated = parsedData.is_automated;                                // (8)
       if (is_automated) {
         badgeData.text[1] = 'automated';                                         // (9)
         badgeData.colorscheme = 'blue';                                          // (9)
       } else {
         badgeData.text[1] = 'manual';                                            // (9)
         badgeData.colorscheme = 'yellow';                                        // (9)
       }
-      badgeData.colorB = '#008bb8';                                              // (9)
+      badgeData.colorB = data.colorB || '#008bb8';                               // (9)
```
Comment (PyvesB) [inline doc/TUTORIAL.md]: Should `makeColorB` be used here?

File: doc/TUTORIAL.md [hunk 2]
```diff
@@ -177,15 +176,19 @@ The following numbering explains what happens in the corresponding lines.
 6. We request the `url` and pass a call back function to the request.
    The function is called once the data is retrieved from the API.
 7. We want to always see a badge regardless the input.
-   In some cases the API may return an error e.g. if the query was invalid.
-   The error is handled and a badge with the status "inaccessible" is returned.
-   ![](https://img.shields.io/badge/docker%20build-inaccessible-lightgrey.svg)
+   In some cases the API may return an error or a HTTP status code indicating
+   a client error or a server error e.g. if the query was invalid. The error response
+   is handled by the [checkErrorResponse](https://github.com/badges/shields/blob/8fcc13d5bced23f53c9f075e51b419060f6cc124/lib/error-helper.js#L8)
+   function and a badge with a appropriate status is returned: "inaccessible"
```
Comment (PyvesB) [inline doc/TUTORIAL.md]: Minor typo, "a*n* appropriate". ^^

File: doc/TUTORIAL.md [hunk 3]
```diff
@@ -121,22 +121,21 @@ cache(function(data, match, sendBadge, request) {
...
-    if (err != null) {                                                           // (7)
-      badgeData.text[1] = 'inaccessible';                                        // (7)
+    if (checkErrorResponse(badgeData, err, res, 'repo not found')) {             // (7)
       sendBadge(format, badgeData);                                              // (7)
       return;                                                                    // (7)
     }
     try {
-      var data = JSON.parse(buffer);                                             // (8)
-      var is_automated = data.is_automated;                                      // (8)
+      var parsedData = JSON.parse(buffer);                                       // (8)
```
Comment (PyvesB) [inline doc/TUTORIAL.md]: Should we use `let` and `const` instead of `var` in this tutorial?

File: doc/TUTORIAL.md (hunk 1 above)
Comment (RedSparr0w) [reply doc/TUTORIAL.md]: Probably a good idea, didn't even realise this was an option, it's currently only used in 5 places.

File: doc/TUTORIAL.md (hunk 2 above)
Comment (platan) [reply doc/TUTORIAL.md]: Thanks, fixed.

File: doc/TUTORIAL.md (hunk 1 above)
Comment (platan) [reply doc/TUTORIAL.md]: This is actually a bug in my opinion. I wanted to update an example with code from `server.js` and fix this bug in a separate PR. 
`colorB` query param can be used to override badge value color. 
This is how it works (000000 is black, it works out of the box):
https://img.shields.io/github/downloads/atom/atom/total.svg?colorB=000000
But it does not work for docker automated badge:
https://img.shields.io/docker/automated/jrottenberg/ffmpeg.svg?colorB=000000
https://img.shields.io/docker/automated/_/ubuntu.svg?colorB=000000
Actually this line:
```js
badgeData.colorB = data.colorB || '#008bb8';
```
overrides colorscheme defined few lines above. 
Overriding is implemented here:
https://github.com/badges/shields/blob/3518b642e7a37f582692cfad0761859da0bf2b2c/lib/make-badge.js#L128
And docker badge always has color `#008bb8`
https://img.shields.io/docker/automated/jrottenberg/ffmpeg.svg
https://img.shields.io/docker/automated/_/ubuntu.svg

File: doc/TUTORIAL.md (hunk 3 above)
Comment (PyvesB) [reply doc/TUTORIAL.md]: Sure! :+1:

File: doc/TUTORIAL.md (hunk 1 above)
Comment (PyvesB) [reply doc/TUTORIAL.md]: Hmmm. Okay, let's deal with this separately.

**Pull request merged**